### ISSUE TITLE
KeyLock Applet Indicator, enhancement ?

### ISSUE THREAD

**Issue by Coeur-Noir** [solus-project/budgie-desktop#1791]
### Current Budgie version
budgie-desktop 10.4 / Ubuntu Budgie 18.04

### Use-cases
As is, the Lock Key Indicator is a very nice applet for people with no built-in LED indicators.

But that applet could also get very friendly for people accessing Budgie desktops remotely ( vnc, anydesk, teamviewer and so on… ) if it was « click-able » to set on/off caps or num ( on the remotely controlled desktop ).

### Proposal
Add set on/off caps or num on click ?

### References
https://discourse.ubuntubudgie.org/t/improvement-for-lock-keys-indicator/1745
https://github.com/adrianiainlam/indicator-keyboard-led

**Comment by Coeur-Noir** [comment]
Since it's possible to use Unity-indicator in Budgie panel, it brought a working solution - see reference links above.

Now if some magic could happen to mix the indicator-keyboard-led's popover menu with Keylock Applet's icons, that would be prettier and amazing…

**Comment by EbonJaeger** [comment]
Budgie already comes with a lock keys indicator applet.

**Comment by Coeur-Noir** [comment]
Yeah I know… but there is no mouse interaction with that Budgie applet : you can't change caps or num state by mouse clicking.

Why want that ? Because it's very very useful when accessing remotely a desktop ( be it Budgie or else… ) through vnc, teamviewer, anydesk, a.s.o…

**Comment by Coeur-Noir** [comment]
Found there another possible enhancement : https://discourse.ubuntubudgie.org/t/lock-keys-indicator-suggestion-option-not-to-show-num-lock-icon/3842

**Comment by JoshStrobl** [comment]
Please, let's not reference the Ubuntu Budgie discourse for comments. If we could include them here that'd be great. It isn't reasonable for me to have to go to a forum to collate suggestions.

**Comment by Coeur-Noir** [comment]
Suggestion is :
⋅ is there a way to add mouse interaction to lock-key applet, for changing state of key ?

This is very useful when remote controlling desktops.

Links provided are examples of working alternatives with an old but still efficient AppIndicator coming from Unity.

**Comment by Coeur-Noir** [comment]
And here @samlane-ma offers a first draft of the feature https://github.com/samlane-ma/toggle-lock-keys

**Comment by JoshStrobl** [comment]
If only people actually contributed changes upstream to us in improving existing applets instead of creating even more third-party ones.

**Comment by Coeur-Noir** [comment]
I think it's the point, it's not another applet, it is lock key « enhanced ».

**Comment by JoshStrobl** [comment]
@Coeur-Noir Except it **is** a separate applet. If it was building on ours, it'd be written in Vala, development occurring in a branch of a fork of budgie-desktop, with an assumption that a PR would be submitted for merge into budgie-desktop proper.

**Comment by Coeur-Noir** [comment]
Ah ok, my bad - forgive me am just a stupid user - and I think Samlane-Ma also worked on a Vala version.

Thank you for your enthusiasm, it's a warm for the heart.

**Comment by JoshStrobl** [comment]
No problem, I just had to mention something because it is quite frustrating to see this constant level of fragmentation rather than contribution to the desktop itself so **all** Budgie users benefit.

**Comment by Coeur-Noir** [comment]
Don't worry it's all very well understood and cared by @fossfreedom over there ! The idea is **not** to fragment.
I just gather information and try to connect good wills as I am unable to « code ».
Sorry for my cold words.

**Comment by samlane-ma** [comment]
@JoshStrobl I should say that I don't mind contributing.  Please note what I made was never intended to be something I released, it was more just something I made quite a while ago when first starting to try making applets, more so just to learn the process.  Fossfreedom's advice was essentially to "do it right" and submit a PR, which I am certainly willing to do.

**Comment by JoshStrobl** [comment]
@samlane-ma That would be much appreciated :heart:

**Comment by samlane-ma** [comment]
PR #2099 
If you are interested in looking at.  Uses xdotool to allow clicking the icons on the panel to toggle the state.  it won't connect the signals if xdotool is not available, as I know it isn't by default on Solus.
